engine/classes/Elgg/Cache/MetadataCache.php

```diff
@@ -227,39 +254,16 @@ public function populateFromEntities(...$guids) {
 		$data = _elgg_services()->metadataTable->getAll($options);
 		elgg_set_ignore_access($ia);
 
-		// make sure we show all entities as loaded
-		foreach ($guids as $guid) {
-			$this->values[$guid] = null;
-			$this->ids[$guid] = null;
-		}
+		$values = [];
 
-		// build up metadata for each entity, save when GUID changes (or data ends)
 		foreach ($data as $i => $row) {
-			$id = $row->id;
-			$name = $row->name;
-			$value = ($row->value_type === 'text') ? $row->value : (int) $row->value;
-			$guid = $row->entity_guid;
-
-			if (isset($this->values[$guid][$name])) {
-				$this->values[$guid][$name] = (array) $this->values[$guid][$name];
-				$this->values[$guid][$name][] = $value;
-			} else {
-				$this->values[$guid][$name] = $value;
-			}
-
-			if (isset($this->ids[$guid][$name])) {
-				$this->ids[$guid][$name] = (array) $this->ids[$guid][$name];
-				$this->ids[$guid][$name][] = $id;
-			} else {
-				$this->ids[$guid][$name] = $id;
-			}
+			$row->value = ($row->value_type === 'text') ? $row->value : (int) $row->value;
+			$metadata = new ElggMetadata($row);
```

Comment:
What do we get for the overhead of creating these objects?

Reply:
Not sure how we would measure it. Definitely better than iterating through bunch of array and querying the DB to compare metadata names and ids

Reply:
We can also just cache attributes